build/squire-raw.js

```diff
@@ -3386,6 +3405,89 @@ var decreaseListLevel = function ( frag ) {
     return frag;
 };
 
+var getTextFromHTMLFragment = function ( self, frag ) {
+    var walker = getBlockWalker( frag ),
+        lines = [],
+        node;
+    while ( node = walker.nextNode() ) {
+        // Strip down to text only
+        lines.push( node.textContent );
+    }
+    return self._doc.createTextNode( lines.join( '\n' ) || '\n' );
+};
+
+var makePreformatted = function ( frag ) {
+    return this.createElement( 'PRE',
+        this._config.tagAttributes.pre, [
+            getTextFromHTMLFragment( this, frag )
+        ] );
+};
+
+var removePreformatted = function ( frag ) {
```

Comment:
OK, this function has many code style violations; I'm not going to list them all. But beyond that, I'm not sure this is necessary at all. Can't you just do (not in a modifyBlocks call) something like:

    expandRangeToBlockBoundaries( range );
    moveRangeBoundariesUpTree( range, this._body );
    proto.removeAllFormatting();

Since your `<pre>` doesn't have any formatting in it anyway, all you really want to do is change it back to the default block type, which is exactly what removeAllFormatting will do.